src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword(user.getPassword(),user.getSalt()));
+        return userDao.add(user);
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        return userDao.get(id);
```

Comment:
but it will be a good strategy to return User object in the service layer, and use orElseThrow method, throwing EntityNotFoundException with info message